lapce-proxy/src/dispatch.rs

```diff
@@ -1265,3 +1227,72 @@ fn git_get_remote_file_url(workspace_path: &Path, file: &Path) -> Result<String>
         file.strip_prefix(workspace_path)?.to_str().unwrap()
     ))
 }
+
+fn start_global_search_job(
+    proxy_rpc: ProxyRpcHandler,
+    global_search_rx: Receiver<()>,
+) {
+    for _ in global_search_rx.iter() {
+        let input = {
+            let mut lock = GLOBAL_SEARCH_INPUT.lock();
+            lock.take()
+        };
+
+        if input.is_none() {
+            continue;
+        }
+
+        let (id, workspace, pattern, case_sensitive) = input.unwrap();
+        let result = if let Some(workspace) = workspace.as_ref() {
+            let mut matches = IndexMap::new();
+            let pattern = regex::escape(&pattern);
+            if let Ok(matcher) = RegexMatcherBuilder::new()
+                .case_insensitive(!case_sensitive)
+                .build_literals(&[&pattern])
+            {
+                let mut searcher = SearcherBuilder::new().build();
+                for path in ignore::Walk::new(workspace).flatten() {
+                    if let Some(file_type) = path.file_type() {
+                        if file_type.is_file() {
+                            let path = path.into_path();
+                            let mut line_matches = Vec::new();
+                            let _ = searcher.search_path(
+                                &matcher,
+                                path.clone(),
+                                UTF8(|lnum, line| {
+                                    let mymatch =
+                                        matcher.find(line.as_bytes())?.unwrap();
+                                    // Shorten the line to avoid sending over absurdly long-lines
+                                    // (such as in minified javascript)
+                                    // Note that the start/end are column based, not absolute from the
+                                    // start of the file.
+                                    line_matches.push((
+                                        lnum as usize,
+                                        (mymatch.start(), mymatch.end()),
+                                        line.chars()
```

Comment:
As with the previous pull request, there is a clear error here.  Regex returns the start of the match in bytes, not characters (https://docs.rs/grep-regex/0.1.10/grep_regex/struct.RegexMatcher.html#method.find).  Take a slice in the range mymatch.start(), mymatch.end(), it will be valid utf-8.

Reply:
And use chars iter after taking the slice.